zjit/src/options.rs

```diff
@@ -163,6 +174,17 @@ fn parse_option(str_ptr: *const std::os::raw::c_char) -> Option<()> {
     match (opt_name, opt_val) {
         ("", "") => {}, // Simply --zjit
 
+        ("mem-size", _) => match opt_val.parse::<usize>() {
+            Ok(n) => {
+                if n == 0 || n > 1024 * 1024 {
```

Comment:
Can you add a comment about what's going on here because it's not clear to me

Reply:
It's not only a check for overflow, but also a nice check (given my version is `n > 1MiB`) to say "you're writing the size in bytes, not in MiB".